CMake: replace usage of the mbed_add_cmake_directory_if_labels() (#13754)
ARMmbed/mbed-os: docs/design-documents/tools/cmake.md

Definitions and configurations would be defined in CMake files, an Mbed app configuration file (`/path/to/app/mbed_app.json`) and Mbed library configuration files (`/path/to/mbed-os/<LIBRARY_NAME>/mbed_lib.json`). `mbed-tools` would parse the Mbed library and app configuration files and generate an `mbed_config.cmake` file.

The following rules must be respected to include special prefix directories (TARGET_, COMPONENT_ and FEATURE_).

Target labels, components, and features defined in `/path/to/mbed-os/targets/targets.json` are used to help the build system determine which directories contain sources/include files to use in the build process. 

> This is a problem as labels, components and features directories will most likely be renamed and will not necessarily use the special prefixes TARGET_, COMPONENT_ and FEATURE_ respectively. Whenever such directories are encountered, check this directory name excluding the special prefix the CMake list `MBED_TARGET_LABELS` generated by `mbed-tools` then include the subdirectory to the build process if found.

Reviewer: ```suggestion
This is a problem as labels, components and features directories will most likely be renamed and will not necessarily use the special prefixes TARGET_, COMPONENT_ and FEATURE_ respectively. Whenever such directories are encountered, check the directory name without the special prefix in the CMake list `MBED_TARGET_LABELS` generated by `mbed-tools` then include the subdirectory to the build process if found.
```

> For example, when the user buildas for the `NUCLEO_F411RE` Mbed target `mbed-tools` includes the `STM` label to the  `MBED_TARGET_LABELS` list. `TARGET_STM` can only be added to the list of build directories as shown below:

Reviewer: ```suggestion
For example when the user builds for the `NUCLEO_F411RE` Mbed target, `mbed-tools` includes the `STM` label to the  `MBED_TARGET_LABELS` list. `TARGET_STM` can only be added to the list of build directories as shown below:
```

```

> # Current working directory: targets/CMakeList.txt

Reviewer: ```suggestion
# CMake input source file: mbed-os/targets/CMakeList.txt
```
